editor/editor_file_system.cpp

```diff
@@ -852,94 +856,96 @@ void EditorFileSystem::_scan_fs_changes(EditorFileSystemDirectory *p_dir, const
 
 		DirAccess *da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
 
-		da->change_dir(cd);
-		da->list_dir_begin();
-		while (true) {
-			String f = da->get_next();
-			if (f == "") {
-				break;
-			}
-
-			if (da->current_is_hidden()) {
-				continue;
-			}
+		Error ret = da->change_dir(cd);
+		if (ret == OK) {
```

Comment:
I just added this `IF`, i don't know why git found all those changes.

Reply:
Indentation change, makes a mess out of the diff.

Reply:
It would look better if you changed `da` to a `DirAccessRef` (so it doesn't need to be `memdelete`d), and then you can do an early return instead of indenting the whole world:
```
Error err = da->change_dir(cd);
if (err) {
    return;
}

da->list_dir_begin();
// etc.
```

Reply:
If we do that, we stop the recursive search of changes on an error.

Here is the code in the bottom of that function
```
	for (int i = 0; i < p_dir->subdirs.size(); i++) {
		if (updated_dir && !p_dir->subdirs[i]->verified) {
			//this directory was removed, add action to remove it
			ItemAction ia;
			ia.action = ItemAction::ACTION_DIR_REMOVE;
			ia.dir = p_dir->subdirs[i];
			scan_actions.push_back(ia);
			continue;
		}
		_scan_fs_changes(p_dir->get_subdir(i), p_progress); // <-- we need to execute this
	}
```

Reply:
Ah right, too bad then.

Reply:
> Ah right, too bad then. Indentation it will be :)

Ok, I analized it a bit further. And I maybe you're right.

For the readers, this function detects changes in the FileSystem and notify Godot on new/delete/modifications. So it doesn't affect the files directly, is just we see on the editor.

And an error on that line is not normal at all. Something very bad happened.

So... If we use what I did every file that has an error, will be treated as `deleted` in Godot File System. So it will 'dissapear' from Godot. So I don't know if that is the desired behavior.

If we add a `return` like you said, it will not update our Godot File System at all and it will remain the same, and we can add a print_error.

If we want to change what I did, I would like to make some tests in the weekend to give more feedback.

Reply:
If an error on this line is not expected you can just use `ERR_FAIL_COND_MSG` and it will return with an error message. It won't process the rest but at that point it might be in a too damaged state for the rest to be meaningful indeed.